### PR TITLE
Add shared storage integration test

### PR DESCRIPTION
**Pull request by acured**
Blocked by NFS mounting issue (maybe port issue). Needs Quanlu's help.

### CODE REVIEW

File: test/nni_test/nnitest/run_tests.py [hunk 1]
```diff
@@ -258,6 +279,11 @@ def run(args):
             print('skipped {}, training service {} not match [{}]'.format(
                 name, args.ts, test_case_config['trainingService']))
             continue
+
+        if not match_remoteConfig(test_case_config):
```
Comment (J-shang) [inline test/nni_test/nnitest/run_tests.py:283]: a little strange if we do not test remote but we need to judge `match_remoteConfig`, maybe put these lines under line 288 is better.

File: test/config/examples/config_sharedstorage_trial.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+"""
+A deep MNIST classifier using convolutional layers.
+
```
Comment (J-shang) [inline test/config/examples/config_sharedstorage_trial.py:3]: If we don't use the nni example, I think create a new folder to handle trial code and config parallel with `test/config/examples` is better.

File: test/nni_test/nnitest/run_tests.py [hunk 3]
```diff
@@ -53,6 +53,18 @@ def update_training_service_config(config, training_service, config_file_path):
         it_ts_config[training_service]['trial']['codeDir'] = containerCodeDir
         it_ts_config[training_service]['trial']['command'] = 'cd {0} && {1}'.format(containerCodeDir, config['trial']['command'])
 
+    if training_service == 'remote':
+        testcase_config = get_yml_content(args.nni_source_dir + config_file_path)
```
[the remark this answers is not in the document]
Comment (acured) [reply test/nni_test/nnitest/run_tests.py]: Fixed

File: test/nni_test/nnitest/run_tests.py [hunk 4]
```diff
@@ -232,6 +244,15 @@ def match_training_service(test_case_config, cur_training_service):
         return True
     return False
 
+def match_remoteConfig(test_case_config):
+    trainingservice_config = get_yml_content(os.path.join('config', 'training_service.yml'))
+    trainingservice_config_reuse_value = str(trainingservice_config['remote']['remoteConfig']['reuse']).lower()
+    testcase_config = get_yml_content(args.nni_source_dir + test_case_config['configFile'])
```
[the remark this answers is not in the document]
Comment (acured) [reply test/nni_test/nnitest/run_tests.py]: Fixed

File: test/config/examples/config_sharedstorage_trial.py (hunk 2 above)
Comment (acured) [reply test/config/examples/config_sharedstorage_trial.py]: Fixed

**Review by J-shang — APPROVED** [review]
I think we need test this in the pipeline before merge.

**Review by SparkSnail — APPROVED** [review]

**Pull request merged**